### DISCUSSION TITLE
Laravel 10 integration

### DISCUSSION THREAD

**Discussion by vojbarzz** [unocss/unocss#2517] [Q&A]
Hello, I'm trying to integrate laravel 10 with unocss using vite. Unfortunately I don't know how to import unocss into my stylesheets.

Here is my `vite.config.js`
```import UnoCSS from 'unocss/vite'
import {defineConfig} from 'vite';
import laravel from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        UnoCSS(),
        laravel({
            input: ['resources/css/app.css', 'resources/js/app.js'],
            refresh: true,
        }),
        // unocssPlugin(),

    ],
});
```
here is my `uno.config.ts`
```
// uno.config.ts
import {defineConfig, presetAttributify, presetUno, presetIcons} from 'unocss'

export default defineConfig({
    presets: [presetUno(), presetAttributify(), presetIcons()],
    rules: [
        ['m-1', { margin: '1px' }]
    ],
})
```

**Comment by cthebest** [comment]
1. To make UnoCSS work correctly with Laravel, you just need to add the following code in the plugins section of vite.config.js:
`UnoCSS({ extraContent: { filesystem: ['./resources/views/**/*.blade.php'], } })`,
This will configure UnoCSS to look for Blade templates in the resources/views directory and any subdirectories when generating CSS.

2. Then we must add the uno.config.js file as specified in the documentation.

3. to add the following line in the app.js file located in resources/js: import 'virtual:uno.css'

4. To finish, add this in your main template: @Vite(['resources/js/app.js']). That should make it work as expected.
